src/modules/tools/endstops/Endstops.cpp

```diff
@@ -679,8 +681,8 @@ void Endstops::home(axis_bitmap_t a)
 
     // Move back a small distance for all homing axis
     this->status = MOVING_BACK;
-    float delta[homing_axis.size()];
-    for (size_t i = 0; i < homing_axis.size(); ++i) delta[i]= 0;
+    float delta[C_AXIS+1];
```

Comment:
This is not an acceptable workaround. Hard coding the size of this array is incorrect. there is a reason it was set to the size of the homing_axis size. This may have unintended consequences.